firewood/src/shale/mod.rs

```diff
@@ -165,12 +165,13 @@ impl<T: Storable> Deref for Obj<T> {
 }
 
 /// User handle that offers read & write access to the stored [ShaleStore] item.
-pub struct ObjRef<'a, T: Storable> {
+#[derive(Debug)]
+pub struct ObjRef<'a, T: Storable + Debug> {
```

Comment:
It would be much better if we could avoid further constraining types themselves (as opposed to implementations). 

If you need to debug the `Drop` implementation, I think it makes sense to just add this code in temporarily while you're debugging.

It's a lot more difficult to remove constraints on types once you rely on them in parent-types.

```suggestion
pub struct ObjRef<'a, T: Storable> {
```

Reply:
+1, makes sense